cudamapper/tests/Test_CudamapperIndexGPUTwoIndices.cu

```diff
@@ -0,0 +1,1333 @@
+    std::vector<SketchElement::DirectionOfRepresentation> expected_directions_of_reads;
+    std::vector<representation_t> expected_unique_representations;
+    std::vector<std::uint32_t> expected_first_occurrence_of_representations;
+
+    expected_representations.push_back(0b0000); // AA(3f1)
+    expected_positions_in_reads.push_back(3);
+    expected_read_ids.push_back(1);
+    expected_directions_of_reads.push_back(SketchElement::DirectionOfRepresentation::FORWARD);
+    expected_unique_representations.push_back(0b00);
+    expected_first_occurrence_of_representations.push_back(0);
+    expected_representations.push_back(0b0000); // AA(4f1)
+    expected_positions_in_reads.push_back(4);
+    expected_read_ids.push_back(1);
+    expected_directions_of_reads.push_back(SketchElement::DirectionOfRepresentation::FORWARD);
+    expected_representations.push_back(0b0010); // AG(5f1)
+    expected_positions_in_reads.push_back(5);
+    expected_read_ids.push_back(1);
+    expected_directions_of_reads.push_back(SketchElement::DirectionOfRepresentation::FORWARD);
+    expected_unique_representations.push_back(0b0010);
+    expected_first_occurrence_of_representations.push_back(2);
+    expected_representations.push_back(0b0100); // CA(2f1)
+    expected_positions_in_reads.push_back(2);
+    expected_read_ids.push_back(1);
+    expected_directions_of_reads.push_back(SketchElement::DirectionOfRepresentation::FORWARD);
+    expected_unique_representations.push_back(0b0100);
+    expected_first_occurrence_of_representations.push_back(3);
+    expected_representations.push_back(0b0101); // CC(1f1)
+    expected_positions_in_reads.push_back(1);
+    expected_read_ids.push_back(1);
+    expected_directions_of_reads.push_back(SketchElement::DirectionOfRepresentation::FORWARD);
+    expected_unique_representations.push_back(0b0101);
+    expected_first_occurrence_of_representations.push_back(4);
+    expected_representations.push_back(0b1001); // GC(0f1)
+    expected_positions_in_reads.push_back(0);
+    expected_read_ids.push_back(1);
+    expected_directions_of_reads.push_back(SketchElement::DirectionOfRepresentation::FORWARD);
+    expected_unique_representations.push_back(0b1001);
+    expected_first_occurrence_of_representations.push_back(5);
+
+    expected_first_occurrence_of_representations.push_back(6);
+
+    test_function(filename,
+                  1, // <- only take second read
+                  2,
+                  minimizer_size,
+                  window_size,
+                  expected_representations,
+                  expected_positions_in_reads,
+                  expected_read_ids,
+                  expected_directions_of_reads,
+                  expected_unique_representations,
+                  expected_first_occurrence_of_representations,
+                  expected_read_id_to_read_name,
+                  expected_read_id_to_read_length,
+                  1);
+}
+
+namespace details
+{
+namespace index_gpu_two_indices
```

Comment:
I'm not sure if gtest executes the tests in definition order, but if so, I would put the details tests first, such that they are the first ones to fail.

Reply:
I was thinking about that, but as google test continues to execute tests even after one test fails I don't think this this is necessary.
Also see this: https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#shuffling-the-tests

Reply:
True, but you usually look the first thing you look at is the first thing that failed, right?

Reply:
Ok. Rearranged.